frontend/test/e2e/admin/datamodel.spec.js

```diff
@@ -0,0 +1,109 @@
+import path from "path";
+
+// FIXME: there are probably better places to put this
+import specReporter from "jasmine-spec-reporter";
+jasmine.getEnv().addReporter(new specReporter);
+
+import { isReady } from "../support/start-server";
+import { setup, cleanup } from "../support/setup";
+import { By, until } from "selenium-webdriver";
+
+import {
+    waitForElement,
+    waitForElementText,
+    waitForElementRemoved,
+    findElement,
+    waitForElementAndClick,
+    waitForElementAndSendKeys,
+    waitForUrl,
+    screenshot,
+    loginMetabase
+} from "../support/utils";
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 600000;
+
+describe("admin/datamodel", () => {
+    let server, sauceConnect, driver;
+
+    beforeAll(async () => {
+        ({ server, sauceConnect, driver } = await setup({
+            name: "admin/datamodel"
+        }));
+    });
+
+    it ("should start", async () => {
+        expect(await isReady(server.host)).toEqual(true);
+    });
+
+    describe("data model editor", () => {
+        it("should allow admin to edit data model", async () => {
+            await driver.get(`${server.host}/`);
+            await loginMetabase(driver, "bob@metabase.com", "12341234");
+            await waitForUrl(driver, `${server.host}/`);
+
+            await driver.get(`${server.host}/admin/datamodel/database`);
+
+            // hide orders table
+            await waitForElementAndClick(driver, ".AdminList-items li:nth-child(2)");
+            await screenshot(driver, "screenshots/admin-datamodel-orders.png");
+
+            await waitForElementAndClick(driver, "#VisibilityTypes span:nth-child(2)");
+            await waitForElementAndClick(driver, "#VisibilitySubTypes span:nth-child(3)");
+
+            // hide fields from people table
+            await waitForElementAndClick(driver, ".AdminList-items li:nth-child(3)");
+
+            await waitForElementAndClick(driver, "#ColumnsList li:first-child .TableEditor-field-visibility");
+            await waitForElementAndClick(driver, ".ColumnarSelector-rows li:nth-child(2) .ColumnarSelector-row");
+
+            await waitForElementAndClick(driver, "#ColumnsList li:nth-child(2) .TableEditor-field-visibility");
+            await waitForElementAndClick(driver, ".ColumnarSelector-rows li:nth-child(3) .ColumnarSelector-row");
+
+            // modify special type for address field
+            await waitForElementAndClick(driver, "#ColumnsList li:first-child .TableEditor-field-special-type");
+            await waitForElementAndClick(driver, ".ColumnarSelector-rows li:nth-child(2) .ColumnarSelector-row");
+
+            //TODO: verify tables and fields are hidden in query builder
```

Comment:
Was this a TODO from the Lewis PR or are we still waiting on this? Is it easy to verify?

Either way, this doesn't block merging for me.